inc/ThirdParty/Plugins/Optimization/Autoptimize.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace WP_Rocket\ThirdParty\Plugins\Optimization;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class Autoptimize implements Subscriber_Interface {
+	/**
+	 * WP Rocket Options instance
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Array containing the errors
+	 *
+	 * @var array
+	 */
+	private $errors = [];
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data $options WP Rocket Options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber listens to.
+	 *
+	 * @return array
+	 * @since  3.10.4
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'admin_notices' => [ 'warn_when_js_aggregation_and_delay_js_active' ],
+		];
+	}
+
+	/**
+	 * Add an admin warning notice when Delay JS and JS Aggregation are both activated.
+	 *
+	 * @since 3.10.4
+	 */
+	public function warn_when_js_aggregation_and_delay_js_active() {
+		if ( ! rocket_get_constant( 'AUTOPTIMIZE_PLUGIN_VERSION', false ) ) {
+			return;
+		}
```

Comment:
I think it would be better adding this check into `get_subscribed_events` method not to register the file events if the plugin is not installed, this will help if we added other events in the future.

Reply:
Initially, I did have it where you are suggesting. However, when `get_subscribed_events()` returns and empty array, that was what seemed to cause the failure in the `removeAllCallbacksExcept()` from detatching the other things from `admin_notices`. I think we will have to leave it here for now, until we can fix the callbacks test method.